### PR TITLE
Feature/members

### PR DESCRIPTION
**Pull request by codesalley**
Allow users to log in and create a post but stay anonymous for visitors who aren't signed in.
technologies used 
- Devise
- bootstrap

### CODE REVIEW

**Comment by descholar-ceo** [comment]
# This Pull Request is under review by [descholar-ceo](https://github.com/descholar-ceo) :eyes: :eyes: 
## Please wait a bit ...

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @juxsalley  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implement to implement all of the minimal requirements for this project, keep it up, and your project is really almost done.
:+1: Linters are properly set and they don't show any error
:+1: You have tried to style the app, this is very good
#### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [x] As a signed-in user you can create new post => When I tried to create a new post it threw the following error on screenshot.

Screenshot from 2021-03-08 11-36-24

- [x]  It is probably caused because of that you are adding the user to the post in a wrong way. Fix it in this migration: 

```ruby
class AddUserIdToPosts < ActiveRecord::Migration[6.1]
  def change
    add_column :posts, :user, :string
  end
end
```
:warning: try `:user_id` instead of `:user`

- [x] User should be able to sign_out ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_out_requirement.png) from Odin)

- [x] As a guest you can see only post's text ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/guest_can_see_only_posts_requirement.png) from Odin)

- [ ] The application has Sign in/Sign out links present in any view of the app ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_out_link_requirement.png) from Pathwright)

- [ ] The postController uses before_action to restrict access to the :new and :create methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from Odin)

- [ ] The application layout has Sign in / Sign out links (So they are not manually repeated in every view) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_out_link_requirement.png) from Pathwright)

- [ ] You have some logic in views: View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb)
  - [ ] There shoud be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright _=> To make sure that I clarify what you need to change on this reqirement: Ensure that there is no if conditions, and no loops in your view (`.html.erb`) files_
  
#### Resources to help you learn about this:

- [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code/)
- [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<div class="mx-auto w-25 ">
+  <% @posts.each do |post| %>
+
+  <div class="list-group d-flex border border-secondary p-4 m-2">
+    <p class="mb-1 fs-5"><%= post.body %></p>
+    <% if user_signed_in? %>
```
Comment (whiz25) [inline app/views/posts/index.html.erb]: - Extract this logic to the `posts_helper` module. This was already suggested in the previous review. Read more [here](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<div class="mx-auto w-25 ">
+  <% @posts.each do |post| %>
```
Comment (whiz25) [inline app/views/posts/index.html.erb]: - put this loop in a partial

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required :pray:
Nice work making the requested changes. As a user, I was able to sign up, create a post and log out. Well done!
Let's make some changes regarding keeping application logic in the views. I left you some comments to help guide you.



### Martin Afani
#### Slack: whiz25

**Review by MahmoudBakr23 — APPROVED** [review]
# **Approved ✔✔**

<hr>

Hello there, great job working on this project and on the required changes by previous reviewers. All seem good to me now. Congratulations :)

You may now merge and submit.

<hr>

Good luck with the next and happy coding 💻👨‍💻

**Pull request merged**